vllm/config.py

```diff
@@ -294,12 +294,19 @@ def compute_hash(self) -> str:
         factors.append(self.quantization)
         factors.append(self.revision)
         factors.append(self.code_revision)
+        factors.append(self.max_model_len)
+        factors.append(self.quantization)
+        factors.append(self.max_logprobs)
+        factors.append(self.disable_sliding_window)
         factors.append(self.trust_remote_code)
+        factors.append(self.mm_processor_kwargs)
+        factors.append(self.generation_config)
+        factors.append(self.model_impl)
+        factors.append(self.override_generation_config)
         factors.append(self.rope_scaling)
         factors.append(self.rope_theta)
-        # rope cos/sin cache depends on the max_position_embeddings
         factors.append(
-            getattr(self.hf_config, "max_position_embeddings", "None"))
+            self.hf_config)  # hf_config can control how the model looks!
```

Comment:
will it potentially contain object id? it's safer if we can convert to some serialization format like json, then we can safely take the whole config into consideration.

Reply:
Yes, it looks like this can contain object id. So we'll need to figure out how to serialize it...

I'm going to make a pass through this PR and remove the items that need object id for now. We only needed one of the config flags set to fix the issue I was looking at

Reply:
Nvm, hf_config is guaranteed to be serializable to json: https://github.com/huggingface/transformers/blob/main/src/transformers/configuration_utils.py#L919 , and it is common for people to save/load it from json.

So what we're doing here is fine.

I will push to_json_string() into the factors list.